npm2deb/templates.py

```diff
@@ -327,6 +327,13 @@
  %(url)s/releases .*/archive/v?([\d\.]+).tar.gz
 """
 
+WATCH['gitlab'] = """version=4
```

Comment:
Hope you have tested this watch expression.
@pravi can you confirm if works fine ?

Reply:
I tested with @gitlab/svgs but it did not work as expected because it used ssh url instead of https url, so we will need to convert ssh url to https.

```
pravi@andhaka:~/forge/debian/git/js-team/@gitlab/svgs/node-gitlab-svgs-1.53.0$ cat debian/watch 
version=3
# It is not recommended use fakeupstream. Please investigate more.
# Origin url: git+ssh://git@gitlab.com/gitlab-org/gitlab-svgs
# Take a look at https://wiki.debian.org/debian/watch/
# See also fakeupstream: http://anonscm.debian.org/viewvc/qa/trunk/cgi-bin/fakeupstream.cgi?view=markup
opts=\
dversionmangle=s/\+(debian|dfsg|ds|deb)(\.\d+)?$//,\
filenamemangle=s/.*=// \
 https://qa.debian.org/cgi-bin/fakeupstream.cgi?upstream=npmjs/@gitlab/svgs .*=svgs-(\d.*)\.(?:tgz|tar\.(?:gz|bz2|xz))

```

Reply:
@pravi can you check `@gitlab/eslint-config` ?

may be we can handle `ssh` repo urls in another PR

Reply:
That is also failing.

```
pravi@andhaka:~/forge/debian/git/js-team/upstream/npm2deb/@gitlab/eslint-config$ cat node-gitlab-eslint-config/debian/watch 
version=4
# It is not recommended use npmregistry. Please investigate more.
# Origin url: git+https://gitlab.com/gitlab-org/gitlab-eslint-config
# Take a look at https://wiki.debian.org/debian/watch/
opts="searchmode=plain,pgpmode=none"  https://registry.npmjs.org/@gitlab/eslint-config https://registry.npmjs.org/@gitlab/eslint-config/-/@gitlab/eslint-config-(\d[\d\.]*)@ARCHIVE_EXT@
```

Reply:
`@` has to be escaped by a regexp in `%(url)s` : `s/\@/\\\@/g'

Reply:
Second field only

Reply:
@pravi: could you retry with last commit ? (python code to review also)

Reply:
```
pravi@andhaka:~/forge/debian/git/js-team/upstream/npm2deb$ python3 npm2deb.py create @gitlab/eslint-config
Traceback (most recent call last):
  File "npm2deb.py", line 7, in <module>
    sys.exit(main(sys.argv))
  File "/home/pravi/forge/debian/git/js-team/upstream/npm2deb/npm2deb/scripts.py", line 157, in main
    args.func(args)
  File "/home/pravi/forge/debian/git/js-team/upstream/npm2deb/npm2deb/scripts.py", line 293, in create
    npm2deb.start()
  File "/home/pravi/forge/debian/git/js-team/upstream/npm2deb/npm2deb/__init__.py", line 83, in start
    self.create_watch()
  File "/home/pravi/forge/debian/git/js-team/upstream/npm2deb/npm2deb/__init__.py", line 199, in create_watch
    args['remodule'] = re.sub(r'\@',r'\\\@',self.name)
NameError: name 're' is not defined

```

You could also try to run npm2deb directly from repo like above.

Reply:
@pravi: seems fixed now

Reply:
It is now downloading the tarball correctly from npm registry, but it is not using gitlab.com repo even though tagis present there. See origin url mentioned in the watch file, I think it is adding a 'git+' to url.
```
cat @gitlab/eslint-config/node-gitlab-eslint-config-1.4.0/debian/watch
version=4
# It is not recommended use npmregistry. Please investigate more.
# Origin url: git+https://gitlab.com/gitlab-org/gitlab-eslint-config
# Take a look at https://wiki.debian.org/debian/watch/
opts="searchmode=plain,pgpmode=none"  https://registry.npmjs.org/@gitlab/eslint-config https://registry.npmjs.org/\@gitlab/eslint-config/-/eslint-config-(\d[\d\.]*)@ARCHIVE_EXT@

```

Reply:
@pravi : it didn't add a 'git+', this was in upstream package (package.json -> repository -> url). 

Regexp added to fix these urls.

Reply:
@pravi: sometimes Gitlab request falls in timeout, then npm2deb falls to npmregistry.

Reply:
@guimard 
with this created watch file
```
version=4
opts=\
dversionmangle=auto,\
opts=filenamemangle=s/.*\/(\d\S+)\/archive\.tar\.gz/<project>-$1\.tar\.gz/g\
 https://gitlab.com/gitlab-org/gitlab-eslint-config/tags?sort=updated_desc .*(\d\S+)/archive\.tar\.gz
```
`test_uscan` gives the warning

```shell
$: uscan --watchfile "debian/watch" --package "node-gitlab-eslist-config" --upstream-version 0 --download-version 1.4.0 --no-download
uscan warn: unrecognized option opts=filenamemangle=s/.*\/(\d\S+)\/archive\.tar\.gz/<project>-$1\.tar\.gz/ghttps://gitlab.com/gitlab-org/gitlab-eslint-config/tags?sort=updated_desc
uscan warn: unknown protocol for LWP: .*(\d\S+)/archive\.tar\.gz
```

download also fails with warning
```shell
$: uscan --download-version 1.4.0
uscan warn: unrecognized option opts=filenamemangle=s/.*\/(\d\S+)\/archive\.tar\.gz/<project>-$1\.tar\.gz/ghttps://gitlab.com/gitlab-org/gitlab-eslint-config/tags?sort=updated_desc
uscan warn: unknown protocol for LWP: .*(\d\S+)/archive\.tar\.gz
```

my `uscan` version is 2.19.2

Reply:
@guimard can't we increase the timeout for gitlab.com?

Reply:
I guess <project-$1> needs to be replaced with the actual project name.